src/extension/extension.ts

```diff
@@ -13,62 +13,341 @@
+export const KAOTO_TESTS_VIEW_REFRESH_COMMAND_ID = 'kaoto.tests.refresh';
+
 export async function activate(context: vscode.ExtensionContext) {
-  console.info("Kaoto Editor extension is alive.");
-
-  const backendI18n = new I18n(backendI18nDefaults, backendI18nDictionaries, vscode.env.language);
-  backendProxy = new VsCodeBackendProxy(context, backendI18n);
-
-  const kieEditorStore = await KogitoVsCode.startExtension({
-    extensionName: "redhat.vscode-kaoto",
-    context: context,
-    viewType: "webviewEditorsKaoto",
-    editorEnvelopeLocator: new EditorEnvelopeLocator("vscode", [
-      new EnvelopeMapping({
-        type: "kaoto",
-        filePathGlob: KAOTO_FILE_PATH_GLOB,
-        resourcesPathPrefix: "dist/webview/editors/kaoto",
-        envelopeContent: {
-          type: EnvelopeContentType.PATH,
-          path: "dist/webview/KaotoEditorEnvelopeApp.js"
-        }
-      }),
-    ]),
-    channelApiProducer: new VSCodeKaotoChannelApiProducer(),
-    backendProxy: backendProxy,
-  });
-
-  vscode.commands.registerCommand('kaoto.open.textualeditor', async() => {
-    if (kieEditorStore.activeEditor !== undefined) {
-      const doc = await vscode.workspace.openTextDocument(kieEditorStore.activeEditor?.document.document.uri);
-      await vscode.window.showTextDocument(doc, vscode.ViewColumn.Beside);
-    }
-  });
-
-  vscode.commands.registerCommand('kaoto.open', (uri: vscode.Uri) => {
-    vscode.commands.executeCommand('vscode.openWith', uri, 'webviewEditorsKaoto');
-  });
-
-  const redhatService = await getRedHatService(context);
-  telemetryService = await redhatService.getTelemetryService();
-  telemetryService.sendStartupEvent();
+	console.info("Kaoto extension is alive."); // TODO switch "all" console logs into Kaoto output channel
+
+	const redhatService = await getRedHatService(context);
+	telemetryService = await redhatService.getTelemetryService();
+	telemetryService.sendStartupEvent();
+
+	const backendI18n = new I18n(backendI18nDefaults, backendI18nDictionaries, vscode.env.language);
+	backendProxy = new VsCodeBackendProxy(context, backendI18n);
+
+	const kieEditorStore = await KogitoVsCode.startExtension({
+		extensionName: "redhat.vscode-kaoto",
+		context: context,
+		viewType: "webviewEditorsKaoto",
+		editorEnvelopeLocator: new EditorEnvelopeLocator("vscode", [
+			new EnvelopeMapping({
+				type: "kaoto",
+				filePathGlob: KAOTO_FILE_PATH_GLOB,
+				resourcesPathPrefix: "dist/webview/editors/kaoto",
+				envelopeContent: {
+					type: EnvelopeContentType.PATH,
+					path: "dist/webview/KaotoEditorEnvelopeApp.js"
+				}
+			}),
+		]),
+		backendProxy: backendProxy,
+		channelApiProducer: new VSCodeKaotoChannelApiProducer(),
+		editorDocumentType: "text" // TODO verify it will not break anything - this is needed for listeners to be able to update Kaoto view sections immediately, look closer on 'kieEditorStore.activeEditor?.startListeningToDocumentChanges()'
```

Comment:
this one worth looking carefully to the Kie documentation or codebase or asking on the kie channels